### ISSUE TITLE
Theme support

### ISSUE THREAD

**Issue by gnuton** [gnuton/asuswrt-merlin.ng#72]
POC
![image](https://user-images.githubusercontent.com/576580/105718266-589a5180-5f21-11eb-830f-08b5fd4201af.png)

**Comment by gnuton** [comment]
Transparent 
![image](https://user-images.githubusercontent.com/576580/108609808-ce82c300-73d0-11eb-8cbc-e137b1ffb754.png)

**Comment by gnuton** [comment]
have you tried to download ssh into the router and download /www ?
if can also work remotely like by copying the original files to /jffs and run a webserver serving them
cp -a /www/ /jffs
cd /jffs/www
/usr/sbin/httpd -i br0 -p 5556
then you from ssh or ftp you can edit the files and you can see the changes in http://localhost:5556

**Comment by redaxmedia** [comment]
@gnuton Not really, could you properly prepare a "clone and serve locally" repository for this? I actually have limited time and cannot spend this fiddling on routers^^

**Comment by gnuton** [comment]
I will think about something..
I think I do need to find a way to serve those files too... I will write
here once i do have a solution for it.

**Comment by redaxmedia** [comment]
Thanks, we keep in touch.

**Comment by gnuton** [comment]
Theming MVP

1. Themes should be stored in 
    /tmp/var/wwwext/themes
2. Themes should have this dir structure
   theme_name/
    +- css/style.css
    +- images/ 
3. Files in this dir will be reachable from
    http://router.asus.com/user/themes/mytheme/css/style.css
4. To enable the theme you have to set up an env var (it will be moved to UI later)
    nvram set custom_theme="mytheme"

**Comment by gnuton** [comment]
wwext requires the files to be copied there at each reboot.
IMHO a better option would be to use /jffs/themes and link it to /www/themes
that can be done adding a line here.
https://github.com/shantanugoel/asus-rt-n66u-merlin/blob/master/release/src/router/others/rootprep.sh

**Comment by gnuton** [comment]
Here I do collect some data on how asuswrt triggers executable from the webpage.

To read external files you have to first downlod them using something like this

1. the JS in the page calls apply.cgi with action=action name (eg: firmware_check)
2. router/web/web.c defines the action targets which create an eventmsg sent to the cfgmnt daemon 
   `using snprintf(event_msg, sizeof(event_msg), HTTPD_GENERIC_MSG, EID_HTTPD_FW_CHECK);`
3. this message is defined in these places
    `
    cfg_mnt/cfg_event.h:	EID_HTTPD_FW_CHECK = 1,
    rc/services.c:		snprintf(event_msg, sizeof(event_msg), HTTPD_GENERIC_MSG, EID_HTTPD_FW_CHECK); 
    httpd/web.c:			snprintf(event_msg, sizeof(event_msg), HTTPD_GENERIC_MSG, EID_HTTPD_FW_CHECK);
    `
4. eval("/usr/sbin/webs_update.sh"); in router/rc/watchdog.c is what launch the process
5. the scripts are in asuswrt-merlin.ng/release/src/router/rom/webs_scripts


then you have to make the content available as asp page as
src/router/www/release_note_amas.asp:﻿<%nvram_dump("release_note.txt","");%>
where release_note.txt is a file in /tmp
